### PR TITLE
TECH-478: Update GET /{GovStackInstance}/{memberClass}/{memberCode}/{applicationCode}/listMethods

### PR DESCRIPTION
**Pull request by KarolinaKopacz**
Need to update tests for /{GovStackInstance}/{memberClass}/{memberCode}/{applicationCode}/listMethods endpoint.

TICKET: https://govstack-global.atlassian.net/browse/TECH-478

### CODE REVIEW

File: test/openAPI/features/list_methods.feature [hunk 1]
```diff
@@ -1,10 +1,29 @@
...
-
-  Request endpoint: GET /{GovStackInstance}/{memberClass}/{memberCode}/{applicationCode}/listMethods
+  @smoke
+  Scenario: Successfully retrieved the list of REST services and endpoints for a service provider
+    Given Wants to retrieve the list of REST services and endpoints for a service provider
+    When User sends GET request with given "serviceId" as serviceId, required path params and header
+    Then User receives a response from the listMethods endpoint
+    And The listMethods endpoint response should be returned in a timely manner
+    And The listMethods endpoint response should have status 200
+    And The listMethods endpoint response should have content-type header
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/list_methods.feature]: Please specify value of the header

File: test/openAPI/features/support/helpers/helpers.js [hunk 2]
```diff
@@ -2,10 +2,16 @@ module.exports = {
...
   getOpenApiEndpoint:
     '{GovStackInstance}/{memberClass}/{memberCode}/{applicationCode}/getOpenAPI',
+  listMethodsEndpoint:
+    '{GovStackInstance}/{memberClass}/{memberCode}/{applicationCode}/listMethods',
   defaultExpectedResponseTime: 15000,
   getOpenApiExpectedSchema: {
     type: 'string',
   },
+  acceptHeader: {
+    key: 'Accept',
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/support/helpers/helpers.js]:
```suggestion
    key: 'content-type',
```

File: test/openAPI/features/list_methods.feature [hunk 3]
```diff
@@ -1,10 +1,29 @@
...
+    And The listMethods endpoint response should match json schema
 
+  @unit @positive
   Scenario: Successfully retrieved the list of REST services and endpoints for a service provider
     Given Wants to retrieve the list of REST services and endpoints for a service provider
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/list_methods.feature:18]: empty line between scenario and given for consistency

File: test/openAPI/features/list_methods.feature [hunk 4]
```diff
@@ -1,10 +1,29 @@
+@method=GET @endpoint=/{GovStackInstance}/{memberClass}/{memberCode}/{applicationCode}/listMethods
 Feature: This endpoint is used to list REST services and endpoints for a service provider.
 
-  List of REST services and endpoints for a service provider
-
-  Request endpoint: GET /{GovStackInstance}/{memberClass}/{memberCode}/{applicationCode}/listMethods
+  @smoke
+  Scenario: Successfully retrieved the list of REST services and endpoints for a service provider
+    Given Wants to retrieve the list of REST services and endpoints for a service provider
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/list_methods.feature:7]: empty line between scenario and given for consistency

File: test/openAPI/features/support/list_methods.js [hunk 5]
```diff
@@ -17,24 +27,73 @@ Given(
...
-      GovStackInstance: 'string',
-      memberClass: 'string',
-      memberCode: 'string',
-      applicationCode: 'string',
-    })
+  'User sends GET request with given {string} as serviceId, required path params and header',
+  serviceId =>
+    specListMethods
+      .get(baseUrl)
+      .withHeaders(acceptHeader.key, acceptHeader.value)
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/support/list_methods.js]: let's just check the response header if it's content-type application/json; charset=utf-8

File: test/openAPI/features/support/list_methods.js [hunk 6]
```diff
@@ -17,24 +27,73 @@ Given(
...
 Then(
-  'The operation returns the list of REST services and endpoints for a service provider',
-  async () => {
-    specListMethods.expectStatus(200).expectJsonSchema(responseSchema);
-    await specListMethods.toss();
+  'The listMethods endpoint response should have content-type header',
+  () => {
+    specListMethods
+      .response()
+      .should.have.header('content-type', acceptHeader.value);
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/support/list_methods.js]:
```suggestion
      .should.have.header(acceptHeader.key, acceptHeader.value);
```
After change in the helpers file

File: test/openAPI/features/list_methods.feature [hunk 7]
```diff
@@ -1,10 +1,29 @@
...
+    When User sends GET request with given "serviceId" as serviceId, required path params and header
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/list_methods.feature]: I'd propose: 
```suggestion
    When User sends GET request with given "serviceId1" as serviceId, "identifier" as GovStackInstance "identifier" as memberClass "alp4aNum3r1c" as memberCode "alp4aNum3r1c" as applicationCode
```
Those values will be rows from 1st row of Examples from next scenario. This way we won't need to write 2 methods corresponding to When in .js file

If it repeats with other feature file we can name it a bit different. Maybe User calls listMethods endpoint with given... ?

File: test/openAPI/features/list_methods.feature [hunk 8]
```diff
@@ -1,10 +1,29 @@
...
+    Then User receives a response from the listMethods endpoint
+    And The listMethods endpoint response should be returned in a timely manner
+    And The listMethods endpoint response should have status 200
+    And The listMethods endpoint response should have content-type header
+    And The listMethods endpoint response should match json schema
 
+  @unit @positive
   Scenario: Successfully retrieved the list of REST services and endpoints for a service provider
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/list_methods.feature]:
```suggestion
  Scenario Outline: Successfully retrieved the list of REST services and endpoints for a service provider
```

**Review by mgrochalskisoldevelo — CHANGES_REQUESTED** [review]

**Review by aleksander-reitsakas — APPROVED** [review]

File: examples/mock/mockoon-bb-information-mediator.json [hunk 9]
```diff
@@ -79,6 +80,20 @@
...
+              "target": "query",
+              "modifier": "serviceId",
+              "value": "^([a-zA-Z][a-zA-Z\\\\d_:\\\\-.]*)$",
+              "invert": false,
+              "operator": "equals"
+            },
+            {
+              "target": "header",
+              "modifier": "content-type",
+              "value": "application/json",
```
Comment (mgrochalskisoldevelo) [inline examples/mock/mockoon-bb-information-mediator.json]: this is not necessary.

File: examples/mock/mockoon-bb-information-mediator.json [hunk 10]
```diff
@@ -79,6 +80,20 @@
               "value": "^[a-zA-Z\\d]+$",
               "invert": false,
               "operator": "regex"
+            },
+            {
+              "target": "query",
+              "modifier": "serviceId",
+              "value": "^([a-zA-Z][a-zA-Z\\\\d_:\\\\-.]*)$",
```
Comment (mgrochalskisoldevelo) [inline examples/mock/mockoon-bb-information-mediator.json]:
```suggestion
              "value": "^([a-zA-Z][a-zA-Z\\d_:\\-.]*)$",
```

Please change regex in mockoon to ```^([a-zA-Z][a-zA-Z\d_:\-.]*)$```

File: test/openAPI/features/list_methods.feature [hunk 11]
```diff
@@ -1,10 +1,31 @@
+@method=GET @endpoint=/{GovStackInstance}/{memberClass}/{memberCode}/{applicationCode}/listMethods
 Feature: This endpoint is used to list REST services and endpoints for a service provider.
 
-  List of REST services and endpoints for a service provider
+  @smoke
+  Scenario: Successfully retrieved the list of REST services and endpoints for a service provider
 
-  Request endpoint: GET /{GovStackInstance}/{memberClass}/{memberCode}/{applicationCode}/listMethods
+    Given Wants to retrieve the list of REST services and endpoints for a service provider
+    When User sends GET request with given "serviceId1" as serviceId, "identifier" as GovStackInstance "identifier" as memberClass "alp4aNum3r1c" as memberCode "alp4aNum3r1c" as applicationCode
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/list_methods.feature]: Now i see that there is ```,``` after ```serviceId```. Maybe:  
```suggestion
    When User sends GET request with given "serviceId1" as serviceId, "identifier" as GovStackInstance, "identifier" as memberClass, "alp4aNum3r1c" as memberCode and "alp4aNum3r1c" as applicationCode
```
or without

File: test/openAPI/features/list_methods.feature [hunk 12]
```diff
@@ -1,10 +1,31 @@
...
+    And The listMethods endpoint response should match json schema
+
+  @unit @positive
+  Scenario Outline: Successfully retrieved the list of REST services and endpoints for a service provider
 
-  Scenario: Successfully retrieved the list of REST services and endpoints for a service provider
     Given Wants to retrieve the list of REST services and endpoints for a service provider
-    When The request to retrieve the list of REST services and endpoints for a service provider is sent
-    Then The operation returns the list of REST services and endpoints for a service provider
+    When User sends GET request with given "serviceId" as serviceId, "<GovStackInstance>" as GovStackInstance "<memberClass>" as memberClass "<memberCode>" as memberCode "<applicationCode>" as applicationCode
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/list_methods.feature]: As previous suggestion

File: test/openAPI/features/support/list_methods.js [hunk 13]
```diff
@@ -17,24 +27,55 @@ Given(
...
+      })
+      .withPathParams({
+        GovStackInstance: GovStackInstance,
+        memberClass: memberClass,
+        memberCode: memberCode,
+        applicationCode: applicationCode,
+      })
 );
 
+Then('User receives a response from the listMethods endpoint', async () => {
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/support/list_methods.js]: unnecessary curly brackets

File: test/openAPI/features/support/list_methods.js [hunk 14]
```diff
@@ -17,24 +27,55 @@ Given(
...
+  await specListMethods.toss();
+});
+
 Then(
-  'The operation returns the list of REST services and endpoints for a service provider',
-  async () => {
-    specListMethods.expectStatus(200).expectJsonSchema(responseSchema);
-    await specListMethods.toss();
+  'The listMethods endpoint response should be returned in a timely manner',
+  () => {
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/support/list_methods.js]: unnecessary curly brackets

File: test/openAPI/features/support/list_methods.js [hunk 15]
```diff
@@ -17,24 +27,55 @@ Given(
...
+    specListMethods
+      .response()
+      .to.have.responseTimeLessThan(defaultExpectedResponseTime);
   }
 );
 
-After(() => {
+Then('The listMethods endpoint response should have status 200', () => {
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/support/list_methods.js]: unnecessary curly brackets

File: test/openAPI/features/support/list_methods.js [hunk 16]
```diff
@@ -17,24 +27,55 @@ Given(
...
+  specListMethods.response().to.have.status(200);
+});
+
+Then(
+  'The listMethods endpoint response should have content-type: application\\/json header',
+  () => {
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/support/list_methods.js]: unnecessary curly brackets

File: test/openAPI/features/support/list_methods.js [hunk 17]
```diff
@@ -17,24 +27,55 @@ Given(
...
+    specListMethods
+      .response()
+      .should.have.header(acceptHeader.key, acceptHeader.value);
+  }
+);
+
+Then('The listMethods endpoint response should match json schema', () => {
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/support/list_methods.js]: unnecessary curly brackets

File: test/openAPI/features/list_methods.feature [hunk 18]
```diff
@@ -1,10 +1,31 @@
+@method=GET @endpoint=/{GovStackInstance}/{memberClass}/{memberCode}/{applicationCode}/listMethods
 Feature: This endpoint is used to list REST services and endpoints for a service provider.
 
-  List of REST services and endpoints for a service provider
+  @smoke
+  Scenario: Successfully retrieved the list of REST services and endpoints for a service provider
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/list_methods.feature]:
```suggestion
  Scenario: Successfully retrieved the list of REST services and endpoints for a service provider smoke type test
```

**Review by mgrochalskisoldevelo — CHANGES_REQUESTED** [review]

**Review by mgrochalskisoldevelo — APPROVED** [review]

**Pull request merged**